fs_storage/models/fs_storage.py

```diff
@@ -363,6 +378,14 @@ def _get_filesystem(self) -> fsspec.AbstractFileSystem:
         directory_path = self.directory_path
         if directory_path:
             fs = fsspec.filesystem("rooted_dir", path=directory_path, fs=fs)
+        if not tools.config["test_enable"]:
+            # Check whether we need to invalidate FS cache or not.
+            # Use a marker file to limit the scope of the LS command for performance.
+            try:
+                self._check_connection(fs)
+            except Exception as e:
+                fs.clear_instance_cache()
+                raise e
```

Comment:
IMO this code should go into the `fs` property method....

Reply:
@sebalix with this change now we ensure the connection is refreshed if needed